### PR TITLE
[TOOLS-3101] Refresh user cache on logout

### PR DESCRIPTION
**Pull request by jhonypg**

### CODE REVIEW

File: core/src/main/java/objective/taskboard/user/UserCacheService.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    private final CacheManager cacheManager;
+
+    @Autowired
+    public UserCacheService(CacheManager cacheManager) {
+        this.cacheManager = cacheManager;
+    }
+
+    public void refreshAllCaches() {
+        String userName = CredentialsHolder.defineUsername();
+        cacheManager.getCacheNames().forEach(cache -> {
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/user/UserCacheService.java]: MINOR Remove useless curly braces around statement [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1602)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: core/src/main/java/objective/taskboard/user/UserCacheService.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            cacheManager.getCache(cache).evict(userName);
```
Comment (herbert-venancio) [inline core/src/main/java/objective/taskboard/user/UserCacheService.java]: This is a very naive approach, what guarantees that no username collides with any other cache key?

Comment (herbert-venancio) [inline core/src/main/java/objective/taskboard/user/UserCacheService.java]: A better solution would be to create a `UserCacheService` that controls how user data gets cached or evicted

File: core/src/main/java/objective/taskboard/auth/CredentialsHolder.java [hunk 3]
```diff
@@ -23,42 +19,44 @@
...
+    private static Authentication getAuthentication() {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        validateAuthentication(authentication);
+        return authentication;
     }
 
-    private static Authentication getAuthentication() {
-        return SecurityContextHolder.getContext().getAuthentication();
+    private static LoggedUserDetails getLoggedInUser() {
+        return (LoggedUserDetails) getAuthentication().getPrincipal();
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/auth/CredentialsHolder.java]: MAJOR A "NullPointerException" could be thrown; "getAuthentication" is nullable here. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2259)

File: core/src/main/java/objective/taskboard/auth/CredentialsHolder.java [hunk 4]
```diff
@@ -23,42 +19,44 @@
...
     public static Optional<String> impersonateUsername() {
-        validateAuthentication();
-        LoggedUserDetails loggedUser = (LoggedUserDetails) getAuthentication().getPrincipal();
-        return loggedUser.getImpersonateUsername();
+        return getLoggedInUser().getImpersonateUsername();
     }
 
     public static String password() {
-        validateAuthentication();
         return (String) getAuthentication().getCredentials();
```
Comment (objective-solutions-user) [inline core/src/main/java/objective/taskboard/auth/CredentialsHolder.java]: MAJOR A "NullPointerException" could be thrown; "getAuthentication" is nullable here. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2259)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: core/src/main/java/objective/taskboard/auth/CredentialsHolder.java [hunk 5]
```diff
@@ -1,7 +1,3 @@
-package objective.taskboard.auth;
-
-import java.util.Optional;
-
 /*-
  * [LICENSE]
```
Comment (herbert-venancio) [inline core/src/main/java/objective/taskboard/auth/CredentialsHolder.java]: Remove [LICENSE]

**Pull request merged**